code/datums/gamemode/role/changeling.dm

```diff
@@ -0,0 +1,599 @@
+
+/datum/power/changeling/boost_range
+	name = "Boost Range"
+	desc = "We evolve the ability to shoot our stingers at humans, with some preperation."
+	genomecost = 2
+	allowduringlesserform = 1
+	verbpath = /obj/item/verbs/changeling/proc/changeling_boost_range
+
+/datum/power/changeling/Epinephrine
+	name = "Epinephrine sacs"
+	desc = "We evolve additional sacs of adrenaline throughout our body."
+	helptext = "Gives the ability to instantly recover from stuns.  High chemical cost."
+	genomecost = 4
+	verbpath = /obj/item/verbs/changeling/proc/changeling_unstun
+
+/datum/power/changeling/ChemicalSynth
+	name = "Rapid Chemical-Synthesis"
+	desc = "We evolve new pathways for producing our necessary chemicals, permitting us to naturally create them faster."
+	helptext = "Doubles the rate at which we naturally recharge chemicals."
+	genomecost = 4
+	isVerb = 0
+	verbpath = /mob/proc/changeling_fastchemical
+/*
+/datum/power/changeling/AdvChemicalSynth
+	name = "Advanced Chemical-Synthesis"
+	desc = "We evolve new pathways for producing our necessary chemicals, permitting us to naturally create them faster."
+	helptext = "Doubles the rate at which we naturally recharge chemicals."
+	genomecost = 8
+	isVerb = 0
+	verbpath = /mob/proc/changeling_fastchemical
+*/
+/datum/power/changeling/EngorgedGlands
+	name = "Engorged Chemical Glands"
+	desc = "Our chemical glands swell, permitting us to store more chemicals inside of them."
+	helptext = "Allows us to store an extra 25 units of chemicals."
+	genomecost = 4
+	isVerb = 0
+	verbpath = /mob/proc/changeling_engorgedglands
+
+/datum/power/changeling/DigitalCamoflague
+	name = "Digital Camouflage"
+	desc = "We evolve the ability to distort our form and proportions, defeating common algorithms used to detect lifeforms on cameras."
+	helptext = "We cannot be tracked by camera while using this skill.  However, humans looking at us will find us... uncanny.  We must constantly expend chemicals to maintain our form like this."
+	genomecost = 3
+	allowduringlesserform = 1
+	verbpath = /obj/item/verbs/changeling/proc/changeling_digitalcamo
+
+/datum/power/changeling/rapidregeneration
+	name = "Rapid Regeneration"
+	desc = "We evolve the ability to rapidly regenerate, negating the need for stasis."
+	helptext = "Heals a moderate amount of damage every tick."
+	genomecost = 8
+	verbpath = /obj/item/verbs/changeling/proc/changeling_rapidregen
+
+/datum/power/changeling/armblade
+	name = "Arm Blade"
+	desc = "We transform one of our arms into an organic blade that can cut through flesh and bone."
+	helptext = "The blade can be retracted by using the same verb used to manifest it. It has a chance to deflect projectiles."
+	genomecost = 5
+	verbpath = /obj/item/verbs/changeling/proc/changeling_armblade
+
+
+/datum/role/changeling/proc/EvolutionMenu()
+	set category = "Changeling"
+	set desc = "Level up!"
+
+	if(!usr || !usr.mind || !usr.mind.GetRole(CHANGELING))
```

Comment:
not a big deal but you might want to only call `usr.mind.GetRole(CHANGELING)` once instead of twice